web/public/js/txadmin/classes.js

```diff
@@ -0,0 +1,221 @@
+      success: function (data) {
+        notify.update("progress", 0);
+        notify.update("type", data.type);
+        notify.update("message", data.message);
+      },
+      error: function (xmlhttprequest, textstatus, message) {
+        notify.update("progress", 0);
+        notify.update("type", "danger");
+        notify.update("message", message);
+      },
+    });
+  }
+
+  // kick player
+  kick() {
+    if (modPlayer.curr.id == false) return;
+    let reason = prompt("Type the kick reason or leave it blank (press enter)");
+    if (reason == null) return;
+
+    var notify = $.notify(
+      { message: '<p class="text-center">Executing Command...</p>' },
+      { type: 'info' }
+    );
+
+    let data = {
+      id: modPlayer.curr.id,
+      reason: reason,
+    };
+    $.ajax({
+      type: "POST",
+      url: "/player/kick",
+      timeout: timeoutLong,
+      data: data,
+      dataType: "json",
+      success: function (data) {
+        notify.update("progress", 0);
+        notify.update("type", data.type);
+        notify.update("message", data.message);
+        if (data.type !== "danger") modPlayer.Modal.hide();
+      },
+      error: function (xmlhttprequest, textstatus, message) {
+        notify.update("progress", 0);
+        notify.update("type", "danger");
+        notify.update("message", message);
+      },
+    });
+  }
+
+  // ban player
+  banPlayer() {
+    const reason = modPlayer.Ban.reason.value.trim();
+  if (!reason.length) {
+    $.notify(
+      { message: '<p class="text-center">The ban reason is required.</p>' },
+      { type: "danger" }
+    );
+    return;
+  }
+  const duration =
+    modPlayer.Ban.durationSelect.value === "custom"
+      ? `${modPlayer.Ban.durationMultiplier.value} ${modPlayer.Ban.durationUnit.value}`
+      : modPlayer.Ban.durationSelect.value;
+
+    const notify = $.notify(
+      { message: '<p class="text-center">Executing Command...</p>' },
+      {}
+    );
+    const data = {
+      reason,
+      duration,
+      reference:
+        modPlayer.curr.id !== false
+          ? modPlayer.curr.id
+          : modPlayer.curr.identifiers,
+      // reference: modPlayer.curr.identifiers,
+    };
+    $.ajax({
+      type: "POST",
+      url: "/player/ban",
+      timeout: timeoutLong,
+      data: data,
+      dataType: "json",
+      success: function (data) {
+        notify.update("progress", 0);
+        notify.update("type", data.type);
+        notify.update("message", data.message);
+        if (data.type !== "danger") modPlayer.Modal.hide();
+      },
+      error: function (xmlhttprequest, textstatus, message) {
+        notify.update("progress", 0);
+        notify.update("type", "danger");
+        notify.update("message", message);
+      },
+    });
+  }
+
+  // warn player
+  warnPlayer() {
+    if (modPlayer.curr.id == false) return;
+    let reason = prompt("Type the warn reason.");
+    if (reason == null) return;
+    reason = reason.trim();
```

Comment:
probably a good idea to retain the logic used before, and we can use the null-conditional operator  (`prompt('x')?.trim()` works for func calls, right?) here if you want to access `trim` when it might be `null`, then just check after if it's `null`